src/libs/antares/study/area/list.cpp

```diff
@@ -907,6 +907,32 @@ static bool AreaListLoadFromFolderSingleArea(Study& study,
             ret = area.hydro.series->loadFromFolder(study, area.id, buffer) && ret;
         }
 
+        if (area.hydro.series && study.header.version < 870)
+        {
+            std::shared_ptr<DataTransfer> datatransfer = std::make_shared<DataTransfer>();
+            buffer.clear() << study.folderInput << SEP << "hydro";
+
+            datatransfer->LoadFromFolder(study, buffer, area);
+            datatransfer->SupportForOldStudies(study, buffer, area);
+        }
+
+        if (area.hydro.series)
+        {
+            buffer.clear() << study.folderInput << SEP << "hydro" << SEP << "series";
+            ret = area.hydro.series->LoadHydroPowerCredits(study, area.id, buffer) && ret;
+        }
+
+        buffer.clear() << study.folderInput << SEP << "hydro" << SEP << "common" << SEP
+                       << "capacity" << SEP << "maxpower_" << area.id << '.'
+                       << study.inputExtension;
+
+        bool exists = IO::File::Exists(buffer);
+
+        if (study.header.version >= 870 && exists)
+        {
+            IO::File::Delete(buffer);
```

Comment:
You are absolutely right. Deleting of deprecated files should be done somewhere else. My proposal is to delete deprecated files just before saving files after upgrading a study. Why before and not after saving files? Because in save function heder version of study will be upgraded and we need to delete files before upgrading study heder version. On these two places corresponding delete function should be:
![image](https://github.com/AntaresSimulatorTeam/Antares_Simulator/assets/131546833/a1e9102f-066d-44a5-a93f-6bef121d5376)

![image](https://github.com/AntaresSimulatorTeam/Antares_Simulator/assets/131546833/3dec8ec5-ca30-4310-a137-92d7b4021895)

Proposal for delete function:

![image](https://github.com/AntaresSimulatorTeam/Antares_Simulator/assets/131546833/be75fe69-dfe9-4644-9843-ebec13a11b5d)